### ISSUE TITLE
Display showcase teaser images better

### ISSUE THREAD

**Issue by ghost** [backdrop-ops/backdropcms.org#623]
On the [Showcase](https://backdropcms.org/showcase) page, the teasers show the first desktop screenshot provided for each site. However, if the screenshot provided was in landscape orientation, then we cannot safely crop it into a square. Then there's a white gap at the bottom which looks off. The shorter the screenshot (in relation to its width), the larger the gap.

Example:
<img width="270" alt="screencapture-backdropcms-org-showcase-2019-12-24-16_12_18 png" src="https://user-images.githubusercontent.com/397895/71427072-949d1c00-2668-11ea-858b-f3eff39095ba.png">

---

PR by BWPanda: https://github.com/backdrop-ops/backdropcms.org/pull/624

**Comment by jenlampton** [comment]
Any suggestions? How about  a gray background where the image is too short? 

<img width="2896" alt="Screen Shot 2019-12-20 at 3 31 44 PM" src="https://user-images.githubusercontent.com/397895/71383417-20437980-2591-11ea-9cc3-d59bd6a384c6.png">

**Comment by jenlampton** [comment]
We could also add more validation on the image field to require a minimum height as well as a minimum width, but I was worried that might prevent people from submitting if they can't get a screenshot at those dimensions that looks great. 

Also, we really only need that validation on the first image uploaded into that field -- it won't matter for any of the other deltas -- and that could be tricky.

**Comment by BWPanda** [comment]
I was thinking just not setting a height (or whatever's causing that gap) and allowing the bottom text to move up directly under the image...

**Comment by jenlampton** [comment]
> allowing the bottom text to move up directly under the image...

That would be worse than what we have now. All the information people read would no longer be in a straight line. For the text, people's eyes move left-to right so it is important that the text to the right is in line with the text to the left. Causing people to scan for the next piece of readable text causes discomfort. 

> The original image is fine

I don't see how the original is fine? The original image is still wider than it is tall.

> the image style used in the Showcase view is setting the height:

The image style crops portrait images to be square, trimming off the bottom. It will not affect the height of landscape images.

**Comment by BWPanda** [comment]
> That would be worse than what we have now.

I disagree. I don't think people come to this page to read the text, and certainly not to read the text of different sites one after another such that if it weren't in a straight line it would be a problem.

I think the screenshots are what people come to see on this page, and the text once the click on a screenshot.

**Comment by jenlampton** [comment]
If that's the case then we'll need a design that can accommodate screenshots that might be provided in landscape orientation. @dariusgarza do you have any recommendations on how to resolve this?

**Comment by dariusgarza** [comment]
How about a staggered masonry layout?

**Comment by BWPanda** [comment]
I see 3 options:

1. Masonry layout as @dariusgarza suggested.
2. Display landscape images unaltered. The text will move up underneath the image and then there'll be a gap under that before the new row of screenshots (a gap below the teaser is better than a gap in the middle of the teaser IMO).
3. Provide a specific field for these images where only square aspect ratios are allowed (or the ability to crop screenshots as squares with help text explaining the need for this).

**Comment by jenlampton** [comment]
Let's proceed with option 1.

**Comment by jenlampton** [comment]
This was rattling around in my head yesterday I was painting the house...  If we are going to use a masonry layout, there's no reason to crop all the other screenshots to be square anymore. It would probably look better if we didn't. Should we change the 512x512 image style to scale but not crop and see how that looks?

Suggested changes for masonry layout:
* change view display from grid to unformatted
* update teaser item CSS (for each breakpoint?)
* Remove CSS height of 512px for the image
* change the 512x512 image style to scale but not crop (?)

**Comment by dariusgarza** [comment]
Sounds good to me, post a screenshot 😊

**Comment by BWPanda** [comment]
Made a PR: https://github.com/backdrop-ops/backdropcms.org/pull/624

Based on https://w3bits.com/css-masonry/ (not perfect, but best CSS-only method I found).

![screencapture-borg-lndo-site-showcase-2019-12-29-14_05_00](https://user-images.githubusercontent.com/2385329/71556821-ae15dd80-2a45-11ea-9769-82a8009bbe41.png)

**Comment by jenlampton** [comment]
Do you think we need fewer items per page now that they are taking up more space?

**Comment by dariusgarza** [comment]
Can we go back to the original design, but subtract the white space created
by the different aspect ratios and have the masonry flow in the gaps? The
truncated snapshots and titles were perfect for thumbnails.

**Comment by BWPanda** [comment]
I'm not sure how to implement @dariusgarza's suggestion... It sounds simple enough: scale all images to 512px wide, then crop any that are taller to 512px as well. However the current image style setup crops short images too, adding the white gap to make the image 512px tall (the white gap was added in https://github.com/backdrop/backdrop-issues/issues/1946, previously it was a black gap).

It seems to me that we need a way to only crop images taller than a certain height, but leave shorter ones alone. I'm surprised this isn't possible (but then maybe I've never needed it before).

The closest issue I can find to this is https://github.com/backdrop/backdrop-issues/issues/2763, but I'm not sure if @jenlampton's proposal would provide this or not.

**Comment by olafgrabienski** [comment]
Would it help to use https://github.com/backdrop-contrib/manualcrop?

**Comment by BWPanda** [comment]
I think it should be automated...

**Comment by BWPanda** [comment]
> Can we go back to the original design, but subtract the white space created by the different aspect ratios and have the masonry flow in the gaps?

I finally worked out a solution for this!

Unfortunately it's not possible in core (trying to crop an image that's smaller than the crop dimensions actually end up increasing the canvas size, producing the ugly grey background we can see in the screenshots above), but there's a module for Drupal that allows cropping an image only if it *exceeds* the given dimensions: [Image max size crop](https://www.drupal.org/project/image_max_size_crop)

I ended up porting this module to Backdrop (https://github.com/backdrop-contrib/image_max_size_crop) and will make a PR tomorrow that adds this module to BackdropCMS.org to address this problem.

**Comment by BWPanda** [comment]
Ok, I've updated the [PR](https://github.com/backdrop-ops/backdropcms.org/pull/624) with a new commit that adds the Image Maximum Crop module to fix this issue and make the Showcase page display as per @dariusgarza's suggestion. Here's what it now looks like:

![Screen Shot 2020-10-07 at 10 16 00-fullpage](https://user-images.githubusercontent.com/2385329/95270332-07d5ee80-0887-11eb-9852-233b310fee19.png)

Basically tall screenshots are cropped to 512px as normal, but now shorter screenshots are left as they are (no cropping).

**Comment by dariusgarza** [comment]
Thanks so much, that's beautiful!♥️👋

**Comment by jenlampton** [comment]
Sounds like that's RTBC! @BWPanda would the maximum image crop module help with profile photos too?

**Comment by BWPanda** [comment]
Thanks everyone! Merged and pushed live: https://backdropcms.org/showcase

**Comment by BWPanda** [comment]
> would the maximum image crop module help with profile photos too?

@jenlampton I don't think so. This module simply prevents cropping of smaller images. The problem with profile photos isn't that small images get cropped, but rather that larger images get cropped in the wrong place.
